mage/query-modules/python/igraphalg.md

```diff
@@ -0,0 +1,258 @@
+| **Edge weights**    | <Highlight color="#FB6E00">**weighted**</Highlight>/<Highlight color="#FB6E00">**unweighted**</Highlight> |
+| **Parallelism**     | <Highlight color="#FB6E00">**sequential**</Highlight>                                                     |
+
+## Procedures
+### `get_all_simple_paths(v, to, cutoff)`
+
+Returns all simple paths in the graph `G` from source to target. A simple path is a path with no repeated nodes.
+
+#### Input:
+
+* `v: Vertex` ➡ Starting node for the path.
+* `to: Vertex` ➡ Ending node for the path.
+* `cutoff: int (default=-1)` ➡ Maximum length of path that is considered. If negative, paths of all lengths are considered.
+
+#### Output:
+
+* `path: List[Vertex]` ➡ List of vertices for a certain path. If there are no paths between the source and target within the given cutoff there is no output.
+
+#### Usage:
+```cypher
+MATCH (n:Label), (m:Label)
+CALL igraphalg.get_all_simple_paths(n, m, 5) YIELD *
+RETURN path;
+```
+
+### `spanning_tree(weights, directed)`
+Returns a minimum spanning tree on a graph `G`.
+ A *minimum spanning tree* is a subset of the edges of a connected, graph that connects all of the vertices without any cycles.
+
+#### Input:
+
+* `weights: string (default=NULL)` ➡  Data key to use for edge weights.
+* `directed: bool (default=False)` ➡  Is graph directed or undirected
+
+#### Output:
+
+* `tree: List[List[Vertex]]` ➡  A minimum spanning tree or forest.
+
+#### Usage:
+
+```cypher
+CALL igraphalg.spanning_tree() 
+YIELD *
+RETURN tree;
+```
+
+### `pagerank(damping, weights, directed,implementation)`
+Returns the PageRank of the nodes in the graph.
+
+PageRank computes a ranking of the nodes in graph G based on the structure of the incoming links. It was originally designed as an algorithm to rank web pages.
+
+#### Input:
+
+* `damping: double (default=0.85)` ➡  Damping parameter for PageRank.
+* `weights: string (default="weight")` ➡  Edge data key to use as a weight. If `None`, weights are set to 1.
+* `directed: bool (default=True)` ➡  Is Graph directed or undirected.
+* `implementation: string (default="prpack")` ➡  Algorithm used for calculating PageRank values. The Algorithm can be either `prpack` or `arpack`.
+
+#### Output:
+
+* `node: Vertex` ➡ Vertex to calculate PageRank for.
+* `rank: double` ➡ Node PageRank.
+
+#### Usage:
+
+```cypher
+CALL igraphalg.pagerank() YIELD *
+RETURN node, rank;
+```
+
+### `get_shortest_path(source, target, weights, directed)`
+Compute the shortest path in the graph.
+
+#### Input:
+
+* `source: Vertex (default=NULL)` ➡  Starting node for the path.
+* `target: Vertex (default=NULL)` ➡  Ending node for the path. 
+* `weights: string (default=NULL)` ➡  If `None`, every edge has weight/distance/cost 1. If a string, use this edge attribute as the edge weight. Any edge attribute not present defaults to 1.
+* `directed: bool (default=True)` ➡  Is Graph directed or undirected.
```

Comment:
```suggestion
* `source: Vertex (default=NULL)` ➡  Path's starting node. 
* `target: Vertex (default=NULL)` ➡  Path's ending node. 
* `weights: string (default=NULL)` ➡  If `None`, every edge has weight/distance/cost 1. If the value is a property name, use that property as the edge weight. If an edge doesn't have a property, the value defaults to 1.
* `directed: bool (default=True)` ➡ If `true`, the graph is directed, otherwise, it's undirected.
```

Reply:
"weights" implies i can list several properties to use as weight?

Reply:
Once again, not involved with this PR. Just noting that in igraph at least, `weights` is plural because the value is a _vector_ of weights, one value for each edge.

Reply:
That's what I thought as well, just wanted to check! Thnx! :)